decode_response.go

```diff
@@ -31,9 +32,27 @@ func (sp *SAMLServiceProvider) ValidateEncodedResponse(encodedResponse string) (
 		return nil, err
 	}
 
-	response, err := sp.validationContext().Validate(doc.Root())
-	if err != nil && !sp.SkipSignatureValidation || response == nil {
-		return nil, err
+	response := doc.Root()
+	if !sp.SkipSignatureValidation {
+		response, err = sp.validationContext().Validate(doc.Root())
+		if err == errors.New("Missing signature referencing the top-level element") {
```

Comment:
After submitting #22, I saw a new test failure caused by conflating a "missing signature" error with any other error that might be returned from `Validate`. Lacking something better, I check for the exact error message.